src/test-table.md

```diff
@@ -100,6 +104,88 @@ for _, tt := range tests {
 }
 ```
 
+## Avoid Unnecessary Complexity in Table Tests
+
+Table tests can be difficult to read and maintain if the subtests contain conditional 
+assertions or other branching logic. Table tests should **NOT** be used whenever 
+there needs to be complex or conditional logic inside subtests (i.e. logic inside the `for` loop).
+
+Large, complex table tests harm readability and maintainability because test readers may
+have difficulty debugging test failures that occur.
+
+Table tests like this should be split into either multiple test tables or multiple 
+individual `Test...` functions.
+
+Some ideals to aim for are:
+
+* Focus on the narrowest unit of behavior
+* Minimize `if` statements and other control flow (if possible)
+* Ensure that all table fields are used in all tests
+* Ensure that all test logic runs for all table cases
+
+Concretely, that means that you should avoid
+having multiple branching pathways (e.g. `shouldError`, `expectCall`, etc.),
```

Comment:
nit: typically it's called `wantErr` (and the example below uses that too)

Reply:
I think the intent was different than `wantErr` (conventionally speaking) - `wantErr` is generally an `error` that you assert (e.g. with `errors.Is`), but `shouldError` is probably a boolean that implies branching for conditional assertion.

Having thought about it a bit more, I think we can probably remove this
paragraph and instead call out the things that are bad in the example below; that would help tie them to a concrete example ("how these things *can be* bad") without necessarily over-prescribing ("these things are *always* bad").